packages/flutter/test/material/page_transitions_theme_test.dart

```diff
@@ -519,4 +524,108 @@ void main() {
     expect(find.text('push'), findsOneWidget);
     expect(find.text('page b'), findsNothing);
   }, variant: TargetPlatformVariant.all());
+
+  testWidgets('ZoomPageTransitionsBuilder uses theme canvasColor during transition effects', (WidgetTester tester) async {
+    final Map<String, WidgetBuilder> routes = <String, WidgetBuilder>{
+      '/': (BuildContext context) => RepaintBoundary(
+        child: Scaffold(
+          appBar: AppBar(
+            title: const Text('Home Page'),
+          ),
+          body: Center(
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: <Widget>[
+                ElevatedButton(
+                  onPressed: () {
+                    Navigator.pushNamed(context, '/scaffolded');
+                  },
+                  child: const Text('Route with scaffold!'),
+                ),
+                ElevatedButton(
+                  onPressed: () {
+                    Navigator.pushNamed(context, '/not-scaffolded');
+                  },
+                  child: const Text('Route with NO scaffold!'),
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+      '/scaffolded': (BuildContext context) => RepaintBoundary(
+        child: Scaffold(
+          appBar: AppBar(
+            title: const Text('Scaffolded Page'),
+          ),
+          body: Center(
+            child: ElevatedButton(
+              onPressed: () {
+                Navigator.pop(context);
+              },
+              child: const Text('Back to home route...'),
+            ),
+          ),
+        ),
+      ),
+      '/not-scaffolded': (BuildContext context) => RepaintBoundary(
+        child: Center(
+          child: ElevatedButton(
+            onPressed: () {
+              Navigator.pop(context);
+            },
+            child: const Text('Back to home route...'),
+          ),
+        ),
+      ),
+    };
+
+    await tester.pumpWidget(
+      RepaintBoundary(
+        child: MaterialApp(
+          theme: ThemeData(
+            canvasColor: const Color.fromARGB(255, 255, 0, 0),
+            primarySwatch: Colors.blue,
+          ),
+          routes: routes,
+        ),
+      ),
+    );
+
+    // Verify initial page
+    expect(find.text('Route with scaffold!'), findsOneWidget);
+    expect(find.text('Route with NO scaffold!'), findsOneWidget);
+
+    // Click to go to scaffolded page
+    await tester.tap(find.text('Route with scaffold!'));
+
+    // Pump till animation is half-way through
+    await tester.pump();
+    await tester.pump(const Duration(milliseconds:75));
+
+    // Use golden file to check canvas color presence
+    await expectLater(
+      find.byType(RepaintBoundary).first,
+      matchesGoldenFile('page_transitions_theme.zoom_page_canvas.scaffolded.png'),
+    );
```

Comment:
As you might know, golden tests are a pain:
- they can be broken by changes not related to this particular test (for instance typography changes).
- the golden infrastructure is sometimes difficult to work with.

We should try to replace those golden tests with something such as:
```dart
    final RenderBox renderBox = ** query the right render box **;
    expect(renderBox, paints..rect(color: expectedColor));
```

Reply:
Makes sense, I'll try to do them using `RenderBox`!

By the way, would you be willing to give your opinion on changing the use of `canvasColor` to `scaffoldBackgroundColor` as discussed above? If two maintainers agree it's a better choice I can get everything out of the way in one sitting.

Reply:
All done, thanks for the suggestion, had no idea you could query `RenderBox ` like that. It even ends up looking much more clean and readable that using golden files!